### ISSUE TITLE
az webapp ssh failed  to setup connection

### ISSUE THREAD

**Issue by marcelkonings** [Azure/azure-cli#9049]
The issue has been copied to your clipboard. Paste it here!
To print out the issue body locally, run `az feedback --verbose`

## Describe the bug
az webapp ssh failed to setup ssh connection

**Command Name**
`az webapp ssh`

**Errors:**

:open_mouth:

```The command failed with an unexpected error. Here is the traceback:

HTTPSConnectionPool(host='ws20availabilityapi100-t.scm.azurewebsites.net', port=443): Max retries exceeded with url: /AppServiceTunnel/Tunnel.ashx?GetStatus (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x111881438>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known'))Traceback (most recent call last):
  File "/usr/local/Cellar/azure-cli/2.0.62/libexec/lib/python3.7/site-packages/urllib3/connection.py", line 159, in _new_conn
    (self._dns_host, self.port), self.timeout, **extra_kw)
  File "/usr/local/Cellar/azure-cli/2.0.62/libexec/lib/python3.7/site-packages/urllib3/util/connection.py", line 57, in create_connection
    for res in socket.getaddrinfo(host, port, family, socket.SOCK_STREAM):
  File "/usr/local/opt/python/Frameworks/Python.framework/Versions/3.7/lib/python3.7/socket.py", line 748, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
socket.gaierror: [Errno 8] nodename nor servname provided, or not known

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/Cellar/azure-cli/2.0.62/libexec/lib/python3.7/site-packages/urllib3/connectionpool.py", line 600, in urlopen
    chunked=chunked)
  File "/usr/local/Cellar/azure-cli/2.0.62/libexec/lib/python3.7/site-packages/urllib3/connectionpool.py", line 343, in _make_request
    self._validate_conn(conn)
  File "/usr/local/Cellar/azure-cli/2.0.62/libexec/lib/python3.7/site-packages/urllib3/connectionpool.py", line 839, in _validate_conn
    conn.connect()
  File "/usr/local/Cellar/azure-cli/2.0.62/libexec/lib/python3.7/site-packages/urllib3/connection.py", line 301, in connect
    conn = self._new_conn()
  File "/usr/local/Cellar/azure-cli/2.0.62/libexec/lib/python3.7/site-packages/urllib3/connection.py", line 168, in _new_conn
    self, "Failed to establish a new connection: %s" % e)
urllib3.exceptions.NewConnectionError: <urllib3.connection.VerifiedHTTPSConnection object at 0x111881438>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/Cellar/azure-cli/2.0.62/libexec/lib/python3.7/site-packages/knack/cli.py", line 206, in invoke
    cmd_result = self.invocation.execute(args)
  File "/usr/local/Cellar/azure-cli/2.0.62/libexec/lib/python3.7/site-packages/azure/cli/core/commands/__init__.py", line 326, in execute
    raise ex
  File "/usr/local/Cellar/azure-cli/2.0.62/libexec/lib/python3.7/site-packages/azure/cli/core/commands/__init__.py", line 384, in _run_jobs_serially
    results.append(self._run_job(expanded_arg, cmd_copy))
  File "/usr/local/Cellar/azure-cli/2.0.62/libexec/lib/python3.7/site-packages/azure/cli/core/commands/__init__.py", line 375, in _run_job
    cmd_copy.exception_handler(ex)
  File "/usr/local/Cellar/azure-cli/2.0.62/libexec/lib/python3.7/site-packages/azure/cli/command_modules/appservice/commands.py", line 54, in _polish_bad_errors
    raise ex
  File "/usr/local/Cellar/azure-cli/2.0.62/libexec/lib/python3.7/site-packages/azure/cli/core/commands/__init__.py", line 354, in _run_job
    result = cmd_copy(params)
  File "/usr/local/Cellar/azure-cli/2.0.62/libexec/lib/python3.7/site-packages/azure/cli/core/commands/__init__.py", line 145, in __call__
    return self.handler(*args, **kwargs)
  File "/usr/local/Cellar/azure-cli/2.0.62/libexec/lib/python3.7/site-packages/azure/cli/core/__init__.py", line 451, in default_command_handler
    return op(**command_args)
  File "/usr/local/Cellar/azure-cli/2.0.62/libexec/lib/python3.7/site-packages/azure/cli/command_modules/appservice/custom.py", line 2528, in ssh_webapp
    create_tunnel_and_session(cmd, resource_group_name, name, port=None, slot=slot)
  File "/usr/local/Cellar/azure-cli/2.0.62/libexec/lib/python3.7/site-packages/azure/cli/command_modules/appservice/custom.py", line 2462, in create_tunnel_and_session
    _wait_for_webapp(tunnel_server)
  File "/usr/local/Cellar/azure-cli/2.0.62/libexec/lib/python3.7/site-packages/azure/cli/command_modules/appservice/custom.py", line 2480, in _wait_for_webapp
    if is_webapp_up(tunnel_server):
  File "/usr/local/Cellar/azure-cli/2.0.62/libexec/lib/python3.7/site-packages/azure/cli/command_modules/appservice/custom.py", line 2435, in is_webapp_up
    return tunnel_server.is_webapp_up()
  File "/usr/local/Cellar/azure-cli/2.0.62/libexec/lib/python3.7/site-packages/azure/cli/command_modules/appservice/tunnel.py", line 92, in is_webapp_up
    preload_content=False
  File "/usr/local/Cellar/azure-cli/2.0.62/libexec/lib/python3.7/site-packages/urllib3/request.py", line 68, in request
    **urlopen_kw)
  File "/usr/local/Cellar/azure-cli/2.0.62/libexec/lib/python3.7/site-packages/urllib3/request.py", line 89, in request_encode_url
    return self.urlopen(method, url, **extra_kw)
  File "/usr/local/Cellar/azure-cli/2.0.62/libexec/lib/python3.7/site-packages/urllib3/poolmanager.py", line 323, in urlopen
    response = conn.urlopen(method, u.request_uri, **kw)
  File "/usr/local/Cellar/azure-cli/2.0.62/libexec/lib/python3.7/site-packages/urllib3/connectionpool.py", line 667, in urlopen
    **response_kw)
  File "/usr/local/Cellar/azure-cli/2.0.62/libexec/lib/python3.7/site-packages/urllib3/connectionpool.py", line 667, in urlopen
    **response_kw)
  File "/usr/local/Cellar/azure-cli/2.0.62/libexec/lib/python3.7/site-packages/urllib3/connectionpool.py", line 667, in urlopen
    **response_kw)
  File "/usr/local/Cellar/azure-cli/2.0.62/libexec/lib/python3.7/site-packages/urllib3/connectionpool.py", line 638, in urlopen
    _stacktrace=sys.exc_info()[2])
  File "/usr/local/Cellar/azure-cli/2.0.62/libexec/lib/python3.7/site-packages/urllib3/util/retry.py", line 398, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='ws20availabilityapi100-t.scm.azurewebsites.net', port=443): Max retries exceeded with url: /AppServiceTunnel/Tunnel.ashx?GetStatus (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x111881438>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known'))
```

## To Reproduce:
Steps to reproduce the behavior. Note: Command arguments have been redacted.

- `Fill in additional info here`
- `Run: az webapp ssh --subscription {} --resource-group {} -n {}`

## Expected Behavior

Getting an ssh connection to the Linux web app environment.

## Environment Summary
```
Darwin-18.5.0-x86_64-i386-64bit
Python 3.7.3
Shell: bash

azure-cli                         2.0.62

```

**Comment by panchagnula** [comment]
@yiliaomsft can you please check & follow up with Ranjith and/OR Patrick.

**Comment by patricklee2** [comment]
@marcelkonings is it possible for you to provide more detailed reproduction steps?

**Comment by patricklee2** [comment]
The issue is due to the scm url being incorrectly generated for ASEs

**Comment by panchagnula** [comment]
@patricklee2 we took a fix for this on remote-connect command on the extension, feel free to move that change to core as well. Also, can you add a test case for this when you make this change - thank you.

**Comment by yugangw-msft** [comment]
fixed by #9098
